torch/testing/_internal/common_cuda.py

```diff
@@ -32,3 +34,52 @@ def initialize_cuda_context_rng():
         for i in range(torch.cuda.device_count()):
             torch.randn(1, device="cuda:{}".format(i))
         __cuda_ctx_rng_initialized = True
+
+
+def tf32_is_not_fp32():
+    if not torch.cuda.is_available():
+        return False
+    if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
+        return False
+    if int(torch.version.cuda.split('.')[0]) < 11:
+        return False
+    return True
+
+
+def tf32_on_and_off(precision=1e-5):
```

Comment:
Instead of `precision`, maybe give this a more descriptive name like `tf32_precision? 

Thinking ahead, do you foresee a need for different precisions for float32 and complex64 with TF32, and/or a need for rtol in addition to atol? I guess this decorator will be applied infrequently enough that it'll be easy to update all instances if you want to change its signature later.

Reply:
Name changed. And yeah, I think that it is infrequently enough, and it will be easy to change this when needed. And I think one precison for complex and float are enough, why not just use the max of the two provisions?